paddle/fluid/operators/mkldnn/prelu_mkldnn_op.cc

```diff
@@ -0,0 +1,161 @@
+/* Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include "paddle/fluid/platform/mkldnn_reuse.h"
+
+namespace paddle {
+namespace operators {
+
+using dnnl::memory;
+using framework::Tensor;
+using platform::GetMKLDNNFormat;
+using platform::MKLDNNDeviceContext;
+using platform::MKLDNNGetDataType;
+using platform::to_void_cast;
+
+template <typename T>
+class PReluMKLDNNHandler
+    : public platform::MKLDNNHandlerT<T, dnnl::prelu_forward,
+                                      dnnl::prelu_backward> {
+ public:
+  PReluMKLDNNHandler(const MKLDNNDeviceContext& dev_ctx,
+                     const mkldnn::engine engine, platform::Place cpu_place,
+                     const Tensor* x, const Tensor* weights,
+                     const std::string& uniq_name)
+      : platform::MKLDNNHandlerT<T, dnnl::prelu_forward, dnnl::prelu_backward>(
+            dev_ctx, engine, cpu_place,
+            platform::CreateKey(dev_ctx, framework::vectorize(x->dims()),
+                                uniq_name)) {
+    if (!this->isCached()) {
+      auto x_md = memory::desc(framework::vectorize(x->dims()),
+                               MKLDNNGetDataType<T>(), x->format());
+      auto weights_md =
+          memory::desc(framework::vectorize(weights->dims()),
+                       MKLDNNGetDataType<T>(), memory::format_tag::any);
+
+      this->AcquireForwardPrimitiveDescriptor(dnnl::prop_kind::forward_training,
+                                              x_md, weights_md);
+      this->AcquireBackwardPrimitiveDescriptor(x_md, weights_md, x_md,
```

Comment:
If you could read is_test value here and based on it skip creation of Backward PD that would be good (to avoid unneccessery allocation during inference)

Reply:
Nice idea